access-control-service-sql-impl/src/main/java/org/eclipse/tractusx/semantics/accesscontrol/sql/model/AccessRulePolicy.java

```diff
@@ -0,0 +1,107 @@
+ *
+ ******************************************************************************/
+
+package org.eclipse.tractusx.semantics.accesscontrol.sql.model;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.eclipse.tractusx.semantics.accesscontrol.api.model.SpecificAssetId;
+import org.eclipse.tractusx.semantics.accesscontrol.sql.model.policy.AccessRulePolicyValue;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import lombok.Data;
+
+@Data
+public class AccessRulePolicy {
+
+   private static final String BPN_RULE_NAME = "bpn";
+   private static final String MANDATORY_SPECIFIC_ASSET_IDS_RULE_NAME = "mandatorySpecificAssetIds";
+   private static final String VISIBLE_SPECIFIC_ASSET_ID_NAMES_RULE_NAME = "visibleSpecificAssetIdNames";
+   private static final String VISIBLE_SEMANTIC_IDS_RULE_NAME = "visibleSemanticIds";
+
+   @JsonProperty( "accessRules" )
+   private Set<AccessRulePolicyValue> accessRules;
+
+   @JsonIgnore
+   public Set<SpecificAssetId> getMandatorySpecificAssetIds() {
+      return accessRules.stream().filter( rule -> MANDATORY_SPECIFIC_ASSET_IDS_RULE_NAME.equals( rule.attribute() ) )
+            .flatMap( rule -> {
+               assertMultiValued( rule, MANDATORY_SPECIFIC_ASSET_IDS_RULE_NAME );
+               return rule.values().stream();
+            } )
+            .map( idValue -> {
+               assertSingleValued( idValue, MANDATORY_SPECIFIC_ASSET_IDS_RULE_NAME + ".*" );
+               return new SpecificAssetId( idValue.attribute(), idValue.value() );
+            } )
+            .collect( Collectors.toSet() );
+   }
+
+   @JsonIgnore
+   public Set<String> getVisibleSpecificAssetIdNames() {
+      return getStringValuesOfRule( VISIBLE_SPECIFIC_ASSET_ID_NAMES_RULE_NAME );
+   }
+
+   @JsonIgnore
+   public Set<String> getVisibleSemanticIds() {
+      return getStringValuesOfRule( VISIBLE_SEMANTIC_IDS_RULE_NAME );
+   }
+
+   @JsonIgnore
+   public String getBpn() {
+      return getStringValueOfRule( BPN_RULE_NAME );
+   }
+
+   @JsonIgnore
+   private Set<String> getStringValuesOfRule( final String ruleName ) {
+      return accessRules.stream().filter( rule -> ruleName.equals( rule.attribute() ) )
+            .flatMap( rule -> {
+               assertMultiValued( rule, ruleName );
+               return rule.values().stream();
+            } )
+            .map(  idValue -> getAccessRulePolicyValueStringFunction( idValue, ruleName ) )
+            .collect( Collectors.toSet() );
+   }
+
+   private String getStringValueOfRule( final String ruleName ) {
+      return accessRules.stream().filter( rule -> ruleName.equals( rule.attribute() ) )
+            .map(  idValue -> getAccessRulePolicyValueStringFunction( idValue, ruleName ) )
+            .findAny().orElse( null );
+   }
+
+   private static String getAccessRulePolicyValueStringFunction( final AccessRulePolicyValue idValue, final String ruleName ) {
```

Comment:
Is there a reason to make the methods static?

Reply:
No reason, it was just something I missed when the IDE extracted the method as static.